kong/db/dao/targets.lua

```diff
@@ -65,7 +65,10 @@ local function clean_history(self, upstream_pk)
     ngx.log(ngx.NOTICE, "[Target DAO] Starting cleanup of target table for upstream ",
                tostring(upstream_pk.id))
     local cnt = 0
-    for _, entry in ipairs(delete) do
+    -- reverse again; so deleting oldest entries first
+    for i = #delete, 1, -1 do
```

Comment:
Great, I noticed this behavior as well and was considering a similar fix, so we are on the same page here :+1: